### PR TITLE
AG-12600 Fix line text clipping to only apply to the line itself

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-12600

See: #2310 and https://github.com/ag-grid/ag-charts/pull/2293#issuecomment-2307178696.

Screenshot 2024-08-29 at 16 27 41

### CODE REVIEW

File: packages/ag-charts-community/src/scene/shape/shape.ts [hunk 1]
```diff
@@ -228,6 +224,51 @@ export abstract class Shape extends Node {
...
+    protected closeClipMask(ctx: CanvasContext) {
+        if (!this.isClipMaskOpen) return;
+
+        ctx.restore();
+        this.isClipMaskOpen = false;
+    }
+
+    override render(renderCtx: RenderContext): void {
+        this.closeClipMask(renderCtx.ctx);
+        super.render(renderCtx);
```
Comment (jacobp100) [inline packages/ag-charts-community/src/scene/shape/shape.ts]:
```suggestion
        this.applyClipMask(renderCtx.ctx);
        super.render(renderCtx);
        this.closeClipMask(renderCtx.ctx);
```

File: packages/ag-charts-community/src/scene/shape/line.ts [hunk 2]
```diff
@@ -92,6 +92,8 @@ export class Line extends Shape implements DistantObject {
             y2 = y;
         }
 
+        this.applyClipMask(ctx);
```
Comment (jacobp100) [inline packages/ag-charts-community/src/scene/shape/line.ts]:
```suggestion
```

**Review by jacobp100 — APPROVED** [review]
Just a suggestion to ensure we balance the start/end clipping calls - but other than that LGTM

**Comment by lsjroberts** [comment]
Changed to being specific to the annotation `CollidableLine` scene as discussed.

**Review by manapeirov — APPROVED** [review]

**Pull request merged**